google-cloud-bigquerystorage/src/test/java/com/google/cloud/bigquery/storage/v1/it/ITBigQueryWriteManualClientTest.java

```diff
@@ -406,6 +409,208 @@ public void testJsonStreamWriterWithDefaultStream()
+    TableInfo tableInfo =
+        TableInfo.newBuilder(tableId, StandardTableDefinition.of(originalSchema)).build();
+    bigquery.create(tableInfo);
+    TableName parent = TableName.of(ServiceOptions.getDefaultProjectId(), DATASET, tableName);
+    WriteStream writeStream =
+        client.createWriteStream(
+            CreateWriteStreamRequest.newBuilder()
+                .setParent(parent.toString())
+                .setWriteStream(
+                    WriteStream.newBuilder().setType(WriteStream.Type.COMMITTED).build())
+                .build());
+
+    // Create test JSON objects
+    JSONObject foo = new JSONObject();
+    foo.put("col1", "aaa");
+    JSONArray jsonArr = new JSONArray();
+    jsonArr.put(foo);
+
+    JSONObject foo2 = new JSONObject();
+    foo2.put("col1", "bbb");
+    JSONArray jsonArr2 = new JSONArray();
+    jsonArr2.put(foo2);
+
+    JSONObject updatedFoo = new JSONObject();
+    updatedFoo.put("col1", "ccc");
+    updatedFoo.put("col2", "ddd");
+    JSONArray updatedJsonArr = new JSONArray();
+    updatedJsonArr.put(updatedFoo);
+
+    // Prepare updated schema
+    Field col2 = Field.newBuilder("col2", StandardSQLTypeName.STRING).build();
+    Schema updatedSchema = Schema.of(ImmutableList.of(col1, col2));
+    TableInfo updatedTableInfo =
+        TableInfo.newBuilder(tableId, StandardTableDefinition.of(updatedSchema)).build();
+
+    // Start writing using the JsonWriter
+    try (JsonStreamWriter jsonStreamWriter =
+        JsonStreamWriter.newBuilder(writeStream.getName(), writeStream.getTableSchema()).build()) {
+      int numberOfThreads = 2;
+      ExecutorService streamTaskExecutor = Executors.newFixedThreadPool(5);
+      CountDownLatch latch = new CountDownLatch(numberOfThreads);
+      // Used to verify data correctness
+      AtomicInteger next = new AtomicInteger();
+      for (int i = 0; i < numberOfThreads; i++) {
+        streamTaskExecutor.submit(
+            () -> {
+              // write 2 rows of aaa on each Thread
+              for (int j = 0; j < 2; j++) {
+                try {
+                  jsonStreamWriter.append(jsonArr);
+                  next.getAndIncrement();
+                } catch (IOException | DescriptorValidationException e) {
+                  e.printStackTrace();
+                }
+              }
+
+              // update TableSchema
+              Table updatedTable = bigquery.update(updatedTableInfo);
+              assertEquals(updatedSchema, updatedTable.getDefinition().getSchema());
+
+              // write filler rows bbb until backend acknowledges schema update due to possible
+              // delay
+              for (int w = 0; w < 20; w++) {
+                ApiFuture<AppendRowsResponse> response2 = null;
+                try {
+                  response2 = jsonStreamWriter.append(jsonArr2);
+                  next.getAndIncrement();
+                } catch (IOException | DescriptorValidationException e) {
```

Comment:
Hopefully with ignore unknown fields, we could keep writing ccc and ddd to the table and expect ddd eventually to show up in the table. This is good for now.

Reply:
Ack. Will keep this in mind.